backend/api/views/tips.py

```diff
@@ -16,19 +21,166 @@ def get_tips():
+        content=data["content"],
+        author=ObjectId(data["user_id"]),
+        posted_time=datetime.now(),
+        latitude=data["latitude"],
+        longitude=data["longitude"],
+        category=data["category"],
+    )
+    tips.save()
+    user = User.objects.get(id=data["user_id"])
+    posted_tips = (user.to_mongo())["posted_tips"]
+    posted_tips.append(ObjectId(tips.id))
+    user.update(posted_tips=posted_tips)
+    return create_response(message="success!")
+
+
+@tips.route("/tips/<tips_id>", methods=["PUT"])
+def edit_tip(tips_id):
+    """
+    PUT function for editing Tips objects
+    """
+    data = request.get_json()
+    tip = Tips.objects.get(id=tips_id)
+    if "title" in data:
+        tip.title = data["title"]
+    if "content" in data:
+        tip.content = data["content"]
+    if "latitude" in data:
+        tip.latitude = data["latitude"]
+    if "longitude" in data:
+        tip.longitude = data["longitude"]
+    if "category" in data:
+        tip.category = data["category"]
+    tip.save()
+    return create_response(message="success!")
+
+
+@tips.route("/tips_votes", methods=["PUT"])
+def change_vote():
+    """
+    PUT function for changing a user's upvote or downvote
+    """
+    data = request.get_json()
+    tip = Tips.objects.get(id=data["tips_id"])
+    if data["vote_type"] == UPVOTE:
+        if ObjectId(data["user_id"]) in tip.to_mongo()["upvotes"]:
+            tip.update(pull__upvotes=ObjectId(data["user_id"]))
+        else:
+            tip_upvotes = tip.to_mongo()["upvotes"]
+            tip_upvotes.append(ObjectId(data["user_id"]))
+            tip.update(upvotes=tip_upvotes)
+            if ObjectId(data["user_id"]) in tip.to_mongo()["downvotes"]:
+                tip.update(pull__downvotes=ObjectId(data["user_id"]))
+
+    if data["vote_type"] == DOWNVOTE:
+        if ObjectId(data["user_id"]) in tip.to_mongo()["downvotes"]:
+            tip.update(pull__downvotes=ObjectId(data["user_id"]))
+        else:
+            tip_downvotes = tip.to_mongo()["downvotes"]
+            tip_downvotes.append(ObjectId(data["user_id"]))
+            tip.update(downvotes=tip_downvotes)
+            if ObjectId(data["user_id"]) in tip.to_mongo()["upvotes"]:
+                tip.update(pull__upvotes=ObjectId(data["user_id"]))
+    return create_response(message="success!")
+
+
+@tips.route("/tips/<tips_id>", methods=["DELETE"])
+def delete_tips_by_id(tips_id):
+    """
+    DELETE function for deleting a tips object by id
+    """
+    for tips in Tips.objects:
+        if tips_id == str(tips.id):
+            user = User.objects.get(id=str(tips.author))
+            posted_tips = (user.to_mongo())["posted_tips"]
+            posted_tips.remove(ObjectId(tips.id))
+            user.update(posted_tips=posted_tips)
+            tips.delete()
+            return create_response(message="success!")
+    return create_response(message="Tip not found")
+
+
+@tips.route("/tips", methods=["DELETE"])

```

Comment:
Just wondering why you dont have a response for this endpoint

Reply:
For the delete all tips endpoint?
I copied the structure of the other delete all endpoints for the other objects and just changed it to tips
There's a clear tips function that creates a response and a helper function that doesn't create a response